lighthouse-core/scripts/update-report-fixtures.js

```diff
@@ -43,15 +46,32 @@ async function update() {
     res(address.port);
   }));
 
+  const oldArtifacts = readArtifacts();
+
   const url = `http://localhost:${port}/dobetterweb/dbw_tester.html`;
   const rawFlags = [
-    '--gather-mode=lighthouse-core/test/results/artifacts',
+    '--gather-mode=' + artifactPath,
     '--throttling-method=devtools',
     url,
   ].join(' ');
   const flags = cliFlags.getFlags(rawFlags);
   await cli.runLighthouse(url, flags, budgetedConfig);
   await new Promise(res => server.close(res));
+
+  if (artifactName) {
+    // Revert everything except the one artifact
+    const newArtifacts = readArtifacts();
+    const finalArtifacts = oldArtifacts;
+    finalArtifacts[artifactName] = newArtifacts[artifactName];
+    fs.writeFileSync(
+      artifactPath + '/artifacts.json',
+      JSON.stringify(finalArtifacts, null, 2) + '\n'
```

Comment:
would it be crazy overkill to use `assetSaver.saveArtifacts(finalArtifacts, artifactPath)` for this? 

`JSON.stringify(artifacts, null, 2)` is the same exact thing we do there, but it's possible that will change or the default file name will change, etc, and we'd get those changes for free here.

Only awkwardness is this would need to set `finalArtifacts.traces = {};` and `finalArtifacts.devtoolsLogs = {};` to keep `saveArtifacts` happy.

Reply:
> would it be crazy overkill to use `assetSaver.saveArtifacts(finalArtifacts, artifactPath)` for this?

oh! We should definitely do this, but we should also do 
`const oldArtifacts = assetSaver.loadArtifacts(artifactPath);`
and
`const newArtifacts = assetSaver.loadArtifacts(artifactPath);`

That would pull in all the devtoolsLogs and traces into `oldArtifacts`, which would then be rewritten to the same files on `assetSaver.saveArtifacts(finalArtifacts, artifactPath)`, so no need to manually revert the traces and whatnot.

(and if, for some unknown reason someone wants to update the sample traces, `	yarn run update:sample-artifacts traces` will do the trick)

All this is untested, but hopefully just works :)

Reply:
> All this is untested, but hopefully just works :)

It does! Thanks, this is much better.